cluster/userdata-master.yaml

```diff
@@ -810,6 +813,67 @@ write_files:
               - kubectl
               - proxy
 
+  - path: /srv/kubernetes/manifests/services/kube-apiserver-readonly-svc.yaml
+    content: |
+      apiVersion: v1
+      kind: Service
+      metadata:
+        name: apiserver-svc
+        namespace: kube-system
+        labels:
+          application: kube-apiserver-ro
+      spec:
+        type: ClusterIP
+        ports:
+          - port: 80
+            protocol: TCP
+            targetPort: 8082
+        selector:
+          application: kube-apiserver
+          version: v1.4.5_coreos.0
+
+  - path: /srv/kubernetes/manifests/deployments/kube-state-metrics.yaml
+    content: |
+      apiVersion: extensions/v1beta1
+      kind: Deployment
+      metadata:
+        name: kube-state-metrics
+        namespace: kube-system
+      spec:
+        replicas: 1
+        template:
+          metadata:
+            labels:
+              app: kube-state-metrics
+          spec:
+            containers:
+            - image: registry.opensource.zalan.do/teapot/kube-state-metrics:v0.3.3
+              name: kube-state-metrics
+              ports:
+              - containerPort: 8080
+              command:
+              - /kube-state-metrics
+              - --port=8080
+              - --apiserver=http://apiserver-svc.kube-system.svc.cluster.local
```

Comment:
A readonly endpoint is required for https://github.com/kubernetes/kube-state-metrics if we can not do this, we have to rethink the setup